lib/rbenv_setup.rb

```diff
@@ -47,10 +55,19 @@ def rbenv_is_a_function?
   end
 
   def setup_shell!
-    unless rbenv_is_a_function?
-      rcfile.open('a') do |f|
-        f.puts %Q|eval "$(rbenv init - #{shell})"|
-      end
+    return if rbenv_is_a_function?
+
+    rcfile.then do |rc|
+      opoo <<~MSG and return unless rc
```

Comment:
Note that, without `rbenv` (and `rbenv-install`, and `ruby-build`) in the path, most other fiddling with Ruby done in this file doesn't actually work. I have a possible fix for that in mind; I'll put up a PR in a bit.